File: base.net/src/main/java/alvin/base/net/ApplicationComponent.java
package alvin.base.net;

import javax.inject.Singleton;

import alvin.base.net.http.HttpModule;
import alvin.base.net.remote.RemoteModule;
import alvin.base.net.socket.SocketModule;
import dagger.Component;
import dagger.android.AndroidInjector;
import dagger.android.support.AndroidSupportInjectionModule;

@Singleton
@Component(modules = {
        AndroidSupportInjectionModule.class,
        ApplicationModule.class,
        HttpModule.class,
        RemoteModule.class,
        SocketModule.class
})
public interface ApplicationComponent extends AndroidInjector<Application> {

    @Component.Builder
    abstract class Builder extends AndroidInjector.Builder<Application> { }
}

File: ui.layout/src/main/java/alvin/ui/layout/ApplicationComponent.kt
package alvin.ui.layout

import alvin.ui.layout.listing.ListingModule
import dagger.Component
import dagger.android.AndroidInjector
import dagger.android.support.AndroidSupportInjectionModule
import javax.inject.Singleton

@Singleton
@Component(modules = [
    AndroidSupportInjectionModule::class,
    ApplicationModule::class,
    ListingModule::class
])
interface ApplicationComponent : AndroidInjector<Application> {

    @Component.Builder
    abstract class Builder : AndroidInjector.Builder<Application>()
}
